workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/NetworkPoliciesValidator.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.validator.impl;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.logger.LoggerTags;
+import io.hyscale.commons.logger.WorkflowLogger;
+import io.hyscale.commons.validator.Validator;
+import io.hyscale.controller.activity.ValidatorActivity;
+import io.hyscale.controller.model.WorkflowContext;
+import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
+import io.hyscale.servicespec.commons.model.service.Agent;
+import io.hyscale.servicespec.commons.model.service.NetworkTrafficRule;
+import io.hyscale.servicespec.commons.model.service.Port;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class NetworkPoliciesValidator implements Validator<WorkflowContext> {
```

Comment:
We also need to add StructuredOutputHandler, in case the user runs the command in background mode.
You can refer to ClusterValidator for its usage. Let me know in case you need to discuss on that.